lib/pbench/server/cache_manager.py

```diff
@@ -185,31 +185,51 @@ def make_cache_object(dir_path: Path, path: Path) -> CacheObject:
 
 
 class Inventory:
-    """Encapsulate the tarfile TarFile object and file stream
+    """Encapsulate the file stream and subprocess.Popen object
 
     This encapsulation allows cleaner downstream handling, so that we can close
     both the extractfile file stream and the tarball object itself when we're
     done. This eliminates interference with later operations.
     """
 
-    def __init__(self, stream: IO[bytes], tarfile: Optional[tarfile.TarFile] = None):
+    def __init__(self, stream: IO[bytes], subproc: subprocess.Popen = None):
         """Construct an instance to track extracted inventory
 
         This encapsulates many byte stream operations so that it can be used
         as if it were a byte stream.
 
         Args:
             stream: the data stream of a specific tarball member
-            tarfile: the TarFile object
+            subproc: the subprocess producing the stream
         """
-        self.tarfile = tarfile
+        self.subproc = subproc
         self.stream = stream
 
     def close(self):
-        """Close both the byte stream and, if open, a tarfile object"""
+        """Close the byte stream and clean up the Popen object"""
+        if self.subproc:
+            if self.subproc.poll() is None:
+                # The subprocess is still running: kill it, drain the outputs,
+                # and wait for its termination.  (If the timeout on the wait()
+                # is exceeded, it will raise subprocess.TimeoutExpired.)
```

Comment:
Which is interesting; as it's called deep within werkzeug after we've returned the `Response` object, I have no idea what this will do... but it's likely a 500 error. Would it be better to catch that (maybe with a log) and avoid propagating the error?

Reply:
To my way of thinking, if this exception gets raised, it's pretty much a panic situation:  we've sent a `SIGKILL` to the subprocess, we've drained the pipes, and it still hasn't gone away after 60 seconds...as far as I can see, there's really nothing left to be done except cry for Mommy....